packages/client/src/workflow-client.ts

```diff
@@ -877,10 +877,14 @@ export class WorkflowClient {
           executionTime: optionalTsToDate(raw.workflowExecutionInfo!.executionTime),
           closeTime: optionalTsToDate(raw.workflowExecutionInfo!.closeTime),
           memo: await decodeMapFromPayloads(this.client.dataConverter, raw.workflowExecutionInfo!.memo?.fields),
-          searchAttributes: mapFromPayloads(
-            searchAttributePayloadConverter,
-            raw.workflowExecutionInfo!.searchAttributes?.indexedFields ?? {}
-          ) as SearchAttributes,
+          searchAttributes: Object.fromEntries(
+            Object.entries(
+              mapFromPayloads(
+                searchAttributePayloadConverter,
+                raw.workflowExecutionInfo!.searchAttributes?.indexedFields ?? {}
+              ) as SearchAttributes
+            ).filter(([_, v]) => v && v.length > 0) // Filter out empty arrays returned by pre 1.18 servers
+          ),
```

Comment:
This breaks clients that expect empty array.

If we leave as is / don't break our side, then people can see it noted as a breaking change in server 1.18 before they upgrade?

Reply:
I'd categorize this as a bug fix, the new behavior is expected.

Reply:
I'd say it's nice/preferable, but not expected, given past behavior and `workflowInfo()`. Would at least add a bang to title

Reply:
Still, if users upgrade the server to 1.18 this is what they'll see.
I'd rather keep the SDK behavior consistent regardless of server version.
I wouldn't consider this breaking.